inc/plugins/class-block-conditions.php

```diff
@@ -263,6 +278,51 @@ public function has_stripe_product( $product ) {
 		return $stripe->check_purchase( $product );
 	}
 
+	/**
+	 * If the block has a hide condition, add the appropriate CSS class.
+	 *
+	 * @param array  $condition Condition.
+	 * @param string $block_content Reference to block content.
+	 * @return string|bool
+	 */
+	public function should_add_hide_css_class( $condition, $block_content ) {
+
+		if ( empty( $condition['type'] ) || empty( $condition['screen_sizes'] ) ) {
+			return false;
+		}
+
+		$screen_sizes = $condition['screen_sizes'];
+		$css_class    = '';
+
+		if ( in_array( 'mobile', $screen_sizes ) ) {
+			$css_class .= ' o-hide-on-mobile';
+		}
+
+		if ( in_array( 'tablet', $screen_sizes ) ) {
+			$css_class .= ' o-hide-on-tablet';
+		}
+
+		if ( in_array( 'desktop', $screen_sizes ) ) {
+			$css_class .= ' o-hide-on-desktop';
+		}
+
+		if ( empty( $css_class ) ) {
+			return false;
+		}
+
+		$class_exists = strstr( $block_content, 'class="' );
```

Comment:
@Soare-Robert-Daniel  Here what happens if the block's parent HTML tags don't have a class but child elements do?

Reply:
It will be put to the first child that has a class.

Reply:
That's not ideal. Is this how Block Visibility is dealing with this too or they've a better solution here? I'm thinking that it can be better with JS but I don't think JS solution would be too appropriate for this.

Reply:
@HardeepAsrani  I that hey inject the class in the first element. If the class attrs does not exist, they will add one.

https://github.com/ndiego/block-visibility/blob/40ead26a5a5b8010f671bb225fa54eecfea05446/includes/frontend/render-block.php#L175-L198

Reply:
I modified the thing to change only the parent node.